server/routes/pubs.js

```diff
@@ -0,0 +1,53 @@
+const express = require('express');
+const router = express.Router();
+// const { ApiService } = require('../services');
+const Pub = require('../database/models/pub');
+
+
+router.get('/data', (req, res, next) => {
+    Pub.find().sort({ 'name': 1 })
+        .then(function (dbPubs) {
+            res.json(dbPubs);
+        })
+    // (error) => res.sendStatus(404)
+    // );
+});
+
+router.post('/', (req, res, next) => {
+    Pub.findOne({ twitterHandle: req.body.twitterHandle }, (err, pub) => {
+        if (err) {
+            console.log("here is the error: ", err)
+        } else if (pub) {
+            alert('That publication is already in the database')
+            window.location.replace('/pub-display');
```

Comment:
I changed this to a `console.log` for now because I cannot remember how to send the error back to the browser (and, apparently, I can't remember how to articulate the question in a way that gets help from Google...).

Reply:
- In the `err` scenario, we should send back a 500 "Internal Server Error".
- In the `pub` scenario, we should send back a 409 "Conflict".
- Below is an MVP implementation. We can get fancy with error handling/messaging another time.

```js
Pub.findOne({...}, (err, pub) => {
  if (err) {
    res.sendStatus(500);
  } else if (pub) {
    res.sendStatus(409);
  } else {
    // create logic
  }
);
```